### PR TITLE
Fix GitHub_3449 test

### PR DESCRIPTION
**Pull request by BruceForstall**
This test was looking for one answer for ARM and ARM64 but another
for other architectures. This didn't work for ARM32 on Windows
(compared to ARM32 on Linux).

Instead, simply allow either answer. This doesn't seem like a
terrible loss of precision, since we already allowed different
answers based on the platform.

### CODE REVIEW

**Comment by BruceForstall** [comment]
Is this an appropriate compromise fix?

See the comments in #3449 and #3655 for context.

@RussKeldorph PTAL
cc @dotnet/jit-contrib

**Comment by BruceForstall** [comment]
Fixes #12493

File: tests/src/JIT/Regression/JitBlue/GitHub_3449/GitHub_3449.cs [hunk 1]
```diff
@@ -15,26 +15,19 @@ public class Program
     // 43e08b2a2c280291, which is different from what the codegen produces. To fix this inconsistency,
     // the constant folding is forced to have the same behavior as the codegen, and the result
     // must be always 43e08b2a2c280290.
+    // 
+    // We don't have a good way to tell if the CLR is compiled by clang or VC++, so we simply allow
+    // both answers.
+
```
Comment (CarolEidt) [inline tests/src/JIT/Regression/JitBlue/GitHub_3449/GitHub_3449.cs]: This seems like the right fix, but the comment above should be changed; I think it's confusing to talk about what RyuJIT codegen does when that is impacted by the compiler that it, in turn, is compiled with.

Comment (BruceForstall) [inline tests/src/JIT/Regression/JitBlue/GitHub_3449/GitHub_3449.cs]: @CarolEidt I agree it is confusing. As such, I don't have a good idea for a fix. Do you have a proposed wording that you would like?

Comment (CarolEidt) [inline tests/src/JIT/Regression/JitBlue/GitHub_3449/GitHub_3449.cs]: Personally, I would just omit a few words; no need to go into the accuracy discussion:
```
    // RyuJIT codegen, VC++, clang and gcc may produce different results for casting uint64 to 
    // double.
    //    1) (double)0x84595161401484A0UL --> 43e08b2a2c280290  (RyuJIT codegen or VC++)
    //    2) (double)0x84595161401484A0UL --> 43e08b2a2c280291  (clang or gcc)
    // Constant folding in RyuJIT simply does (double)0x84595161401484A0UL in its C++ implementation.
    // If it is compiled by clang or gcc, the example unsigned value and cast tree node are folded into 
    // 43e08b2a2c280291, which is different from what RyuJIT codegen or VC++ produces.
```
and then finish with the comment you added above.

Comment (BruceForstall) [inline tests/src/JIT/Regression/JitBlue/GitHub_3449/GitHub_3449.cs]: Thanks. I'll use that.

**Pull request merged**